conan/internal/rest/remote_manager.py

```diff
@@ -281,15 +282,19 @@ def _call_remote(self, remote, method, *args, **kwargs):
             raise ConanException(exc, remote=remote)
 
 
-def uncompress_file(src_path, dest_folder, scope=None):
+def uncompress_file(src_path, dest_folder, scope=None, cache_folder=None):
     try:
         filesize = os.path.getsize(src_path)
         big_file = filesize > 10000000  # 10 MB
         if big_file:
             hs = human_size(filesize)
             ConanOutput(scope=scope).info(f"Decompressing {hs} {os.path.basename(src_path)}")
-        with open(src_path, mode='rb') as file_handler:
-            tar_extract(file_handler, dest_folder)
+
+        compression_plugin = load_compress_plugin(cache_folder)
```

Comment:
The plugin cannot be loaded for the de-compression of every file. It should be loaded just once by the ``ConfigAPI`` likely, then used here.